monitor.go

```diff
@@ -0,0 +1,36 @@
+package nuki
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/go-ble/ble"
+)
+
+type StateChangeHandler func(context.Context, *Client)
+
+// MonitorStateChanges listens to advertisements from the given devices and triggers the callback
+// once their state changes. The client - which must be paired with ClientIdTypeBridge - can then
+// call ReadStates to fetch the new state and reset the flag.
+func MonitorStateChanges(ctx context.Context, clb StateChangeHandler, clients ...*Client) error {
```

Comment:
The name "MonitorStateChanges" sounds to me that this method will permanently monitor devices. But you only call  once the ble.Scan method. Will the ble.Scan() scan infinity until the context is close? Or does it only one "scan-cycle" :thinking:

Reply:
The `ble.Scan()` method will keep scanning until you cancel the context. The `MonitorStateChange` function intentionally has the same behavior - it's up to the user to pass a context with a timeout or explicitly cancel it on a certain event.

Reply:
Ok than it should be reflected in comment and/or in the example. Maybe you can change the example with an context with timeout:
```
....
ctx, cancelFn := context.WithTimeout(context.Background(), 10*time.Second)
defer cancelFn()
	
MonitorStateChanges(ctx, handleStateChange, nukiClient)
....
```